android/jni/app/organicmaps/Map.cpp

```diff
@@ -22,147 +22,156 @@ void OnRenderingInitializationFinished(std::shared_ptr<jobject> const & listener
 
 extern "C"
 {
-using namespace storage;
+JNIEXPORT jboolean JNICALL
+Java_app_organicmaps_Map_nativeCreateEngine(JNIEnv * env, jclass,
+                                            jobject surface, jint density,
+                                            jboolean firstLaunch,
+                                            jboolean isLaunchByDeepLink,
+                                            jint appVersionCode)
+{
+  return g_framework->CreateDrapeEngine(env, surface, density, firstLaunch, isLaunchByDeepLink,
+                                        base::asserted_cast<uint32_t>(appVersionCode));
+}
 
-JNIEXPORT void JNICALL
-Java_app_organicmaps_MapFragment_nativeCompassUpdated(JNIEnv * env, jclass clazz, jdouble north, jboolean forceRedraw)
+JNIEXPORT jboolean JNICALL
+Java_app_organicmaps_Map_nativeIsEngineCreated(JNIEnv * env, jclass)
 {
-  location::CompassInfo info;
-  info.m_bearing = north;
+  return g_framework->IsDrapeEngineCreated();
+}
 
-  g_framework->OnCompassUpdated(info, forceRedraw);
+JNIEXPORT jboolean JNICALL
+Java_app_organicmaps_Map_nativeShowMapForUrl(JNIEnv * env, jclass, jstring url)
+{
+  return g_framework->ShowMapForURL(jni::ToNativeString(env, url));
 }
 
 JNIEXPORT void JNICALL
-Java_app_organicmaps_MapFragment_nativeStorageConnected(JNIEnv * env, jclass clazz)
+Java_app_organicmaps_Map_nativeSetRenderingInitializationFinishedListener(
+  JNIEnv * env, jclass, jobject listener)
 {
-  android::Platform::Instance().OnExternalStorageStatusChanged(true);
-  g_framework->AddLocalMaps();
+  if (listener)
+  {
+    g_framework->NativeFramework()->SetGraphicsContextInitializationHandler(
+      std::bind(&OnRenderingInitializationFinished, jni::make_global_ref(listener)));
+  }
+  else
+  {
+    g_framework->NativeFramework()->SetGraphicsContextInitializationHandler(nullptr);
+  }
 }
 
-JNIEXPORT void JNICALL
-Java_app_organicmaps_MapFragment_nativeStorageDisconnected(JNIEnv * env, jclass clazz)
+JNIEXPORT jboolean JNICALL
+Java_app_organicmaps_Map_nativeAttachSurface(JNIEnv * env, jclass, jobject surface)
 {
-  android::Platform::Instance().OnExternalStorageStatusChanged(false);
-  g_framework->RemoveLocalMaps();
+  return g_framework->AttachSurface(env, surface);
 }
 
 JNIEXPORT void JNICALL
-Java_app_organicmaps_MapFragment_nativeScalePlus(JNIEnv * env, jclass clazz)
+Java_app_organicmaps_Map_nativeDetachSurface(JNIEnv * env, jclass, jboolean destroySurface)
```

Comment:
```suggestion
Java_app_organicmaps_Map_nativeDetachSurface(JNIEnv *, jclass, jboolean destroySurface)
```

Please clean up other cases too.